### PR TITLE
[MODORDERS-1174] Use CQL query to filter pieces based on user affiliations

### PR DESCRIPTION
**Pull request by Saba-Zedginidze-EPAM**
## Purpose
[[MODORDERS-1174] Filter pieces based on user affiliations](https://folio-org.atlassian.net/browse/MODORDERS-1174)

## Approach
- Replace filtering with CQL query after fetching user tenants first
- Add SettingsRetriever to filter pieces only when central ordering setting is enabled
- Update tests

### CrossModulesApiTest
image

### OrdersApiTest
image

### ConsortiaOrdersApiTest
image

### CODE REVIEW

File: src/main/java/org/folio/service/pieces/PieceStorageService.java [hunk 1]
```diff
@@ -187,6 +179,15 @@ private Future<List<Piece>> getPieceChunkByLineIds(Collection<String> poLineIds,
       .map(PieceCollection::getPieces);
   }
 
+  static String getQueryForUserTenants(List<String> userTenants, String query) {
+    if (userTenants == null) {
+      return query;
+    }
+    userTenants.add(null);
+    var cql = convertIdsToCqlQuery(userTenants, "receivingTenantId");
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/pieces/PieceStorageService.java:187]: If we have tenantId length for example 20 symbols, the calculation suggests that approximately 103 strings receivingTenantId=<<tenantId>>, each with a length of 20 characters, can be included in GET request without exceeding the 4KB URL limit.
For the largest consortia we have 61 tenant, so we should have some capacity for the original 'query'

Comment (SerhiiNosko) [reply src/main/java/org/folio/service/pieces/PieceStorageService.java:187]: but to be on more safe side we can add checking setting into settings table(because not all consortia can use central ordering): ALLOW_ORDERING_WITH_AFFILIATED_LOCATIONS
Here is how this setting is set on UI: https://folio-dev-thunderjet-consortium.ci.folio.org/settings/consortia-settings/central-ordering
This also can speedup the sql query because this query would be more simple without checking receivingTenantId
This setting should use cache, it can not be disabled when it was enabled

Comment (Saba-Zedginidze-EPAM) [reply src/main/java/org/folio/service/pieces/PieceStorageService.java:196]: Thanks, now filters based on the setting

File: src/main/java/org/folio/service/pieces/PieceStorageService.java [hunk 2]
```diff
@@ -134,21 +137,19 @@ public Future<PieceCollection> getAllPieces(int limit, int offset, String query,
...
-      .filter(piece -> Optional.ofNullable(piece.getReceivingTenantId())
-        .map(userTenants::contains)
-        .orElse(true))
-      .toList();
+      .compose(consortiumConfiguration -> {
+        if (consortiumConfiguration.isEmpty()) {
+          return Future.succeededFuture();
+        }
+        var configuration = consortiumConfiguration.get();
+        return settingsRetriever.getSettingByKey(SettingKey.CENTRAL_ORDERING_ENABLED, requestContext)
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/pieces/PieceStorageService.java:147]: Module descriptors of endpoints that use getPieces should be updated with the new module permission

**Review by SerhiiNosko — APPROVED** [review]

File: src/main/java/org/folio/service/pieces/PieceStorageService.java [hunk 3]
```diff
@@ -187,9 +190,19 @@ private Future<List<Piece>> getPieceChunkByLineIds(Collection<String> poLineIds,
...
-  private static boolean shouldFilterPiecesForTenant(String centralTenantId, RequestContext requestContext) {
+  static String getQueryForUserTenants(List<String> userTenants, String query) {
+    if (CollectionUtils.isEmpty(userTenants)) {
+      return query;
+    }
+    var cql = convertIdsToCqlQuery(userTenants, "receivingTenantId");
+    var matchNullTenantExpr = getCqlExpressionForFieldNullValue("receivingTenantId");
+    cql = combineCqlExpressions("or", cql, matchNullTenantExpr);
+    return combineCqlExpressions("and", query, cql);
+  }
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/pieces/PieceStorageService.java:201]: this part is a little bit hard to follow. Suggest:
```
static String getQueryForUserTenants(List<String> userTenants, String query) {
    if (CollectionUtils.isEmpty(userTenants)) {
        return query;
    }

    String tenantCql = convertIdsToCqlQuery(userTenants, "receivingTenantId");
    String nullTenantCql = getCqlExpressionForFieldNullValue("receivingTenantId");
    String combinedTenantCql = combineCqlExpressions("or", tenantCql, nullTenantCql);

    return combineCqlExpressions("and", query, combinedTenantCql);
}
```

File: src/main/java/org/folio/service/settings/util/SettingFields.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+  private final String value;
+
+  SettingFields(String value) {
+    this.value = value;
+  }
+
+  public String getValue() {
+    return value;
+  }
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/settings/util/SettingFields.java:15]: we can use Lombok in here if possible
@Getter
@RequiredArgsConstructor

File: pom.xml [hunk 5]
```diff
@@ -588,6 +588,10 @@
           <showWeaveInfo>true</showWeaveInfo>
           <forceAjcCompile>true</forceAjcCompile>
           <aspectDirectory>src/main/java/org/folio/rest/annotations</aspectDirectory>
+          <includes>
+            <include>**/impl/*.java</include>
+            <include>**/*.aj</include>
+          </includes>
```
Comment (Saba-Zedginidze-EPAM) [inline pom.xml:594]: Copied from mod-orders-storage as building with lombok annotations was failing

File: src/main/java/org/folio/service/pieces/PieceStorageService.java (hunk 3 above)
Comment (Saba-Zedginidze-EPAM) [reply src/main/java/org/folio/service/pieces/PieceStorageService.java]: Both comments addressed

**Review by azizbekxm — APPROVED** [review]
